File: 알고리즘 문제풀이/프로그래머스/스킬트리.py
def solution(skill,skill_trees):
    answer=0
    for tree in skill_trees:
        mine,idx=[],0
        for t in tree:
            if t not in skill:
                mine.append(t)
            elif t in skill:
                if skill[idx]!=t:
                    break
                else:
                    mine.append(skill[idx])
                    idx+=1
        if len(mine)==len(tree):
            answer+=1
    return answer

File: 알고리즘 문제풀이/프로그래머스/스킬트리.js
function solution(skill, skill_trees) {
  let answer = 0;
  skill_trees.map((tree) => {
    let mine = "",
      idx = 0;
    for (let i = 0; i < tree.length; i++) {
      if (skill.include(tree.charAt(i))) {
        if (skill.charAt(idx) === tree.charAt(i)) {
          mine += tree.charAt(i);
          idx += 1;
        } else {
          break;
        }
      } else {
        mine += tree.charAt(i);
      }
    }
    if (mine.length === tree.length) answer += 1;
  });
  return answer;
}
